### DISCUSSION TITLE
Tailwind breakpoints not working

### DISCUSSION THREAD

**Discussion by janderson2024** [tailwindlabs/tailwindcss#11847] [Help]
What version of Tailwind CSS are you using?

 tailwind v3.3.2

What build tool (or framework if it abstracts the build tool) are you using?

NextJs 13.4.10 - app directory

What version of Node.js are you using?

Node.js v20.5.1

What browser are you using?

Chrome

What operating system are you using?

windows 10

Reproduction URL:
https://swipe-n-hire-axd4b131y-janderson2024.vercel.app/test

Example Video:
https://github.com/tailwindlabs/tailwindcss/assets/34382737/f9265f49-964f-4b13-a53c-8a1f796f029a

Describe your issue:

Put simple I have a page that looks like this: 
`export default function TestPage(){

    return (
        <>
            <h1 className="hidden sm:block">Hi!</h1>
            <h1 className="sm:hidden">Bye!</h1>
        </>

    );
}`

Expected behavior is that _Hi!_ would be visable until the small breakpoint then _Bye!_ would be visable. As seen by the attached video, this doesn't happen. 

Included is my tailwind.config.js
`/** @type {import('tailwindcss').Config} */

import { withUt } from "uploadthing/tw";

module.exports = {
  content: [
    './src/pages/**/*.{js,ts,jsx,tsx,mdx}',
    './src/components/**/*.{js,ts,jsx,tsx,mdx}',
    './src/app/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  theme: {
    extend: {
      spacing: {
        'main-under-nav': 'calc(100vh - 64px)',
        'no-scrollbar': 'calc(100% - 20px)'
      }
    },
  },
  plugins: [],
};' 

That file is the only one we have modified for our project.

After looking through other discussions: we are not doing anything with custom imports in the globals.css, and we have set the viewport meta tag already. 

Anything else needed to help figure out this problem can be provided, and any help is appreciated!

**Comment by wongjn** [comment]
You seem to have two sets of Tailwind CSS rules generated in the CSS. This then causes "mismatched" ordering of the rules in the CSS. For example, you have `hidden` rule at column 17360, but `sm:block` at column 13667 and `sm:hidden` at 13715. In a "normal" Tailwind compilation, `sm:block` and `sm:hidden` would be generated *after* `hidden`: https://play.tailwindcss.com/55RbFRhQyI

**Reply by janderson2024** [reply]
I fully agree this is the issue, but how do I prevent the hidden rule from generating after? I tried switching them around in the component, but that doesn't change the generation order

**Reply by wongjn** [reply]
It seems like some misconfiguration in the project. You should only have one Tailwind-generated stylesheet but it seems you might have several concatenated together.

**Comment by janderson2024** [comment]
Here is another Reproduction URL with all of the breakpoints before and after
https://swipe-n-hire-4vu741zyr-janderson2024.vercel.app/test

**Comment by Ikdemm** [comment]
Any news on this problem? I'm getting the same issue with Chrome but not with Firefox

**Reply by wongjn** [reply]
Consider providing a [Tailwind Play](https://play.tailwindcss.com) or a project that reproduces the unexpected behavior.
